src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -168,6 +169,73 @@ class StatementSegment(ansi_dialect.get_segment("StatementSegment")):  # type: i
     )
 
 
+@snowflake_dialect.segment()
+class ClusterBySegment(BaseSegment):
+    """A `CLUSTER BY` segment."""
+
+    type = "clusterby_segment"
+    # https://docs.snowflake.com/en/user-guide/tables-clustering-keys.html#defining-a-clustering-key-for-a-table
+
+    match_grammar = StartsWith(
```

Comment:
I think this should just be `Sequence` rather than `StartsWith`. Currently this expression could consume a lot more than intended.